packages/cli/src/lib/get-oxygen-token.ts

```diff
@@ -0,0 +1,88 @@
+import {renderConfirmationPrompt} from '@shopify/cli-kit/node/ui';
+import {
+  outputContent,
+  outputInfo,
+  outputToken,
+  outputWarn,
+} from '@shopify/cli-kit/node/output';
+
+import {linkStorefront} from '../commands/hydrogen/link.js';
+import {login} from './auth.js';
+import {getCliCommand} from './shell.js';
+import {getConfig} from './shopify-config.js';
+import {renderMissingLink, renderMissingStorefront} from './render-errors.js';
+import {getOxygenToken} from './graphql/admin/oxygen-token.js';
+
+interface Arguments {
+  root: string;
+  /**
+   * Optional shop override that developers would have passed using the --shop
+   * flag.
+   */
+  flagShop?: string;
+  /**
+   * Does not prompt the user to fix any errors that are encountered (e.g. no
+   * linked storefront)
+   */
+  silent?: boolean;
+}
+
+export async function getOxygenDeploymentToken({
+  root,
+  silent,
+}: Arguments): Promise<string | undefined> {
+  const [{session, config}, cliCommand] = await Promise.all([
+    login(root),
+    getCliCommand(),
+  ]);
+  let configStorefront = (await getConfig(root)).storefront;
+
+  if (!configStorefront?.id) {
+    if (!silent) {
```

Comment:
When is `silent` used? I remember we had this pattern somewhere else but we have removed it. If it's not used now, let's remove it here as well.